### PR TITLE
update compressor.py for 2018

### PR DESCRIPTION
**Pull request by ariovistus**

### CODE REVIEW

File: wpilib/wpilib/compressor.py [hunk 1]
```diff
@@ -148,14 +152,34 @@ def getSmartDashboardType(self):
...
-        table = self.getTable()
-        if table is not None:
-            table.putBoolean("Enabled", self.enabled())
-            table.putBoolean("Pressure Switch", self.getPressureSwitchValue())
+        if self.enabledEntry is not None:
+            self.enabledEntry.setBoolean(self.enabled())
+        if self.pressureSwitchEntry is not None:
+            self.pressureSwitchEntry.setBoolean(self.getPressureSwitchValue())
+
+    def startLiveWindowMode(self):
```
Comment (virtuald) [inline wpilib/wpilib/compressor.py]: Hm? Isn't this done in liveWindowSendable?

Comment (ariovistus) [reply wpilib/wpilib/compressor.py]: what? setting up the listener? wrong entry

**Comment by ariovistus** [comment]
added a bunch of tests (and found some bugs). hal is breaking on test_compressor_clearallpcmstickyfaults, but it's kind of opaque, not sure if it's a problem with Compressor, tests, or hal

File: wpilib/wpilib/compressor.py (hunk 1 above)
Comment (auscompgeek) [reply wpilib/wpilib/compressor.py]: Well, only one entry is on the LiveWindow here. Simple fix.

Comment (ariovistus) [reply wpilib/wpilib/compressor.py]: ?

**Comment by virtuald** [comment]
`clearAllPCMStickyFaults` was implemented incorrectly in the simulated HAL. Examining the declaration functions.py shows that it has no return value, and the simulated function was returning `False`.

**Pull request merged**